src/main/java/net/sf/jabref/util/MassSetFieldAction.java

```diff
@@ -150,7 +164,7 @@ public void stateChanged(ChangeEvent e) {
             public void actionPerformed(ActionEvent e) {
                 // Check if the user tries to rename multiple fields:
                 if (rename.isSelected()) {
-                    String[] fields = getFieldNames(field.getText());
+                    String[] fields = getFieldNames((String) field.getSelectedItem());
```

Comment:
Yeah, it is by design as arbitrary objects can be inserted if the ComboBox is editable: http://stackoverflow.com/a/7026724/873282

Solution: Replace the code by 

    field.getItemAt(field.getSelectedIndex())

(or add a comment pointing to the stackoverflow answer :innocent:)